src/bin.ts

```diff
@@ -251,7 +252,21 @@ function getCwd (dir?: string, scriptMode?: boolean, scriptPath?: string) {
       throw new TypeError('Script mode cannot be combined with `--dir`')
     }
 
-    return dirname(scriptPath)
```

Comment:
I think I’d be more comfortable noting that we only support symlinked directories instead of doing this, but up to you. If we only support directories it would become easy. Alternatively we can implement the resolver in a loop ourselves instead of messing with require.

Reply:
We might not even have TSX, JSX support extensions added either, so it’s another difference here. You could expose the function that creates a list of extensions, then just loop over that list and resolve like node.js does since we don’t need the extra require logic.

Reply:
Do we need to support loading a directory containing an `index.*` file, or a directory containing a `package.json` with a `main` field pointing to another file?  E.g. `require('./directory-containing-package-json')`?

The chicken-and-egg problem with file extensions is a real bummer.  I'll try to come up with some realistic examples of this causing problems in real life.

The only reason I was hoping to support symlinked scripts is so they can be put on the `PATH`.  `npm` and `yarn` do this a lot, and I can see it coming up in monorepo situations. (lerna, yarn2)